File: Practica General Auxiliatura/INTRODUCCION A POO/Eje1/eje1.py
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad

    def saludo(self):
        return f"Hola, soy {self.nombre} de {self.ciudad}"

    def es_mayor_de_edad(self):
        return self.edad >= 18

personas = [
    Persona("Juan", 15, "La Paz"),
    Persona("Ana", 17, "El Alto"),
    Persona("Luis", 25, "Cochabamba")
]
for p in personas:
    print(p.saludo())
    print("Mayor de edad:", p.es_mayor_de_edad())

File: Practica General Auxiliatura/INTRODUCCION A POO/Eje1/eje1.java
class Persona {
    String nombre;
    int edad;
    String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    public String saludo() {
        return "Hola, soy " + this.nombre + " de " + this.ciudad;
    }

    public boolean esMayorDeEdad() {
        return this.edad >= 18;
    }

    public static void main(String[] args) {
        Persona[] personas = {
            new Persona("Juan", 15, "La Paz"),
            new Persona("Ana", 17, "El Alto"),
            new Persona("Luis", 25, "Cochabamba")
        };

        for (Persona p : personas) {
            System.out.println(p.saludo());
        }
    }
}
